bigcodebench/sanitize.py

```diff
@@ -211,38 +252,26 @@ def script(
 
     new_solutions = []
 
-    for solution in tqdm(load_solutions(samples)):
-        task_id = solution["task_id"]
-        if task_id not in dataset:
-            print(
-                f"Skiping {task_id} as it does not existing in the latest EvalPlus dataset."
-            )
-            continue
-
-        function_name = entry_point[task_id] if task_id in entry_point else None
-        dbg_identifier = solution["_identifier"]
-        if debug_task is not None and task_id != debug_task:
-            continue
-
-        ntotal += 1
-        if "solution" in solution:
-            old_code = solution["solution"]
-            if calibrate:
-                old_code = solution["solution"].replace("```python\n    ", "```python\n"+dataset[task_id]["complete_prompt"]+"    ")
-        else:
-            assert "completion" in solution
-            old_code = dataset[task_id]["complete_prompt"] + "\n" + solution["completion"]
-
-        new_code = sanitize(code=old_code, entrypoint=function_name)
-        # if changed, print the message
-        if new_code != old_code:
-            msg = "Sanitized: " + dbg_identifier
-            if is_folder:
-                msg += " -> " + dbg_identifier.replace(samples, target_path)
-            print(msg)
+    parallel_arg_list = [
+        {
+            "sample_solution": sample_solution,
+            "dataset": dataset,
+            "entry_point": entry_point,
+            "debug_task": debug_task,
+            "calibrate": calibrate,
+            "is_folder": is_folder,
+            "target_path": target_path
+        }
+        for sample_solution in load_solutions(samples)
+    ]
+
+    results = pqdm(parallel_arg_list, process_solution, n_jobs=min(parallel, os.cpu_count()), argument_type="kwargs")
```

Comment:
Are there any deprecation concerns related to `pqdm`? To me, `pqdm` is not very actively maintained ATM. Is `pqdm` considered a common practice?

In `evaluate`, we have a [customized parallel execution process[(https://github.com/bigcode-project/bigcodebench/blob/19baeb475f379b535932255a71b3578cda2d04a5/bigcodebench/evaluate.py#L51).

Reply:
pqdm is a convenient wrapper around concurrent.futures with progress bar functionality, very useful for tracking large scale parallelization